### DISCUSSION TITLE
Probability of remaining event free above 1 as time progresses with Gradient boosting (with coxph loss).

### DISCUSSION THREAD

**Discussion by Harisk96** [sebp/scikit-survival#357] [Q&A]
Hello,

I am running a gradient boosting model with repeated stratified Kfold to tune parameters for a gradient boosted model with coxph as loss function. When returning the array and as time progresses, at the later time points I get probability of survival for some samples above 1, and for some values I get an infinite small value "inf". 

Is there any simple explanation behind this? 
I reckon that the beta coefficients might get abnormally large possibly when predicting the survival function for cox?

**Comment by sebp** [comment]
Is this with ComponentwiseGradientBoostingSurvivalAnalysis or 
GradientBoostingSurvivalAnalysis?

If you could even provide a minimal working example that illustrates the 
problem, that would be greatly appreciated.

**Reply by Harisk96** [reply]
Sorry, I should probably have replied here, I attached an example in a new comment below :)

**Comment by Harisk96** [comment]
[CWGDBoost_rskf.pdf](https://github.com/sebp/scikit-survival/files/11434878/CWGDBoost_rskf.pdf)
I happens for both the component wise gradient boosting and the gradient boosting survival analysis algorithms, as well as for Coxnet.

I attached a PDF of an example with the breast cancer dataset from sci-kit survival, as I am not allowed to upload the data I am currently using online. I stopped at a fold having this case, and printed the return array. I have attached a screenshot as well of the "return array" of the survival curve prediction, where we can see some values becoming "inf" and some values getting pretty large, such as "3.511301e+48".

This does not happen for all folds, only for some folds in the repeated stratified k-fold.

This issue seem to predominantly occur when number of estimators are above 400, but might in some cases occur below that.

As well when running those problematic folds, I get this particular warning from sci-kit survival source code:
/opt/anaconda3/lib/python3.9/site-packages/sksurv/linear_model/coxph.py:126: RuntimeWarning: overflow encountered in power
  y=np.power(self.baseline_survival_.y, risk_score[i]))



![Skjermbilde 2023-05-09 kl  20 46 05](https://github.com/sebp/scikit-survival/assets/54271356/862fd630-e77b-4be0-beef-252d53d8b348)

**Comment by Harisk96** [comment]
![Skjermbilde 2023-05-09 kl  20 50 43](https://github.com/sebp/scikit-survival/assets/54271356/657d7c4b-914e-4dd0-8251-20ba584cb93c)

Which in turn might result in a survival curve such as this one for particular folds.
Here i took a 5 sample extract from that same fold.

**Reply by sebp** [reply]
It seems that risk scores get too large or small such the exponential results in an over- or underflow. It might help to rescale the features with sklearn's StandardScaler or similar before fitting the model.

**Reply by Harisk96** [reply]
Yes exactly! I did try rescaling the features using both standardscaler, powertransform and min_max scaling. And min_max scaling seems to have the issue for fewer splits than the others.
